security-jwt/src/main/java/io/micronaut/security/token/jwt/cookie/JwtCookieClearerLogoutHandler.java

```diff
@@ -39,46 +40,76 @@
      */
     @Deprecated
     public JwtCookieClearerLogoutHandler(JwtCookieConfiguration jwtCookieConfiguration) {
-            this.jwtCookieConfiguration = jwtCookieConfiguration;
-            this.logout = jwtCookieConfiguration.getLogoutTargetUrl();
+        this.jwtCookieConfiguration = jwtCookieConfiguration;
+        this.accessTokenCookieConfiguration = jwtCookieConfiguration;
+        this.refreshTokenCookieConfiguration = null;
+        this.logout = jwtCookieConfiguration.getLogoutTargetUrl();
     }
 
     /**
      * @param jwtCookieConfiguration JWT Cookie Configuration
      * @param redirectConfiguration Redirect configuration
      */
-    @Inject
+    @Deprecated
     public JwtCookieClearerLogoutHandler(JwtCookieConfiguration jwtCookieConfiguration,
                                          RedirectConfiguration redirectConfiguration) {
         this.jwtCookieConfiguration = jwtCookieConfiguration;
+        this.accessTokenCookieConfiguration = jwtCookieConfiguration;
+        this.refreshTokenCookieConfiguration = null;
+        this.logout = redirectConfiguration.getLogout();
+    }
+
+    /**
+     * @param accessTokenCookieConfiguration JWT Cookie Configuration
+     * @param refreshTokenCookieConfiguration Refresh token cookie configuration
+     * @param redirectConfiguration Redirect configuration
+     */
+    @Inject
+    public JwtCookieClearerLogoutHandler(AccessTokenCookieConfiguration accessTokenCookieConfiguration,
+                                         RefreshTokenCookieConfiguration refreshTokenCookieConfiguration,
+                                         RedirectConfiguration redirectConfiguration) {
+        this.accessTokenCookieConfiguration = accessTokenCookieConfiguration;
+        this.refreshTokenCookieConfiguration = refreshTokenCookieConfiguration;
+        this.jwtCookieConfiguration = null;
         this.logout = redirectConfiguration.getLogout();
     }
 
     @Override
     public MutableHttpResponse<?> logout(HttpRequest<?> request) {
-        Optional<Cookie> maybeCookie = request.getCookies().findCookie(jwtCookieConfiguration.getCookieName());
         try {
             URI location = new URI(logout);
-            if (maybeCookie.isPresent()) {
-                Cookie requestCookie = maybeCookie.get();
-                String domain = jwtCookieConfiguration.getCookieDomain().orElse(null);
-                String path = jwtCookieConfiguration.getCookiePath().orElse(null);
-                Cookie responseCookie = Cookie.of(requestCookie.getName(), "");
-                responseCookie.maxAge(0).domain(domain).path(path);
-                return HttpResponse.seeOther(location).cookie(responseCookie);
+            MutableHttpResponse<?> response = HttpResponse.seeOther(location);
+            clearCookie(accessTokenCookieConfiguration, request, response);
+            if (refreshTokenCookieConfiguration != null) {
```

Comment:
if we set 

`this.refreshTokenCookieConfiguration = jwtCookieConfiguration;`
instead of: 

`this.refreshTokenCookieConfiguration = null;`

this if check will not be necessary.